### ISSUE TITLE
False positive no-name-in-module(E0611) in module nh3

### ISSUE THREAD

**Issue by jasperfirecai2** [pylint-dev/pylint#8756]
### Bug description

![image](https://github.com/pylint-dev/pylint/assets/8398131/16834843-b369-42b8-a971-7da07c46b985)
Intellisense can resolve this module import but pylint claims this not to not exist. [nh3](https://github.com/messense/nh3) has no pylint-plugin as it shouldn't need one. [issue on their repo](https://github.com/messense/nh3/issues/18)

### Command used

```shell
pylint.exe --reports=n --output-format=json --rcfile=.pylintrc --from-stdin [...]/a.py
```
via pylint vscode plugin 
OR
```shell
python -m pylint --rcfile=.pylintrc $(git ls-files '*.py')
```

### Pylint output

```shell
************* Module a
a.py:19:0: E0611: No name 'clean' in module 'nh3' (no-name-in-module)
```

### Expected behavior

There should be no linter warning here since the module resolves in both intellisense and code.

### Pylint version
Legacy version because vscode plugin used to be broken
```shell
pylint 2.13.4
astroid 2.11.7
Python 3.9.1
```
Also applies to newer version
```shell
pylint 2.17.4
astroid 2.15.5
Python 3.9.1
```

### OS / Environment

Windows 10 Enterprise 64-bit
vscode builtin terminal

### Additional dependencies

for specific issue:
nh3==0.2.13

within project:
pylint-django==2.5.3
pylint-plugin-utils==0.8.2

**Comment by mbyrnepr2** [comment]
Thanks @jasperfirecai2. I think this is the same situation as https://github.com/pylint-dev/pylint/issues/6281.
nh3 is Rust but it does have a pyi file which we could potentially start checking.

**Comment by jasperfirecai2** [comment]
nh3 is a wrapper for ammonia, which is written in rust aye

**Comment by Zedifuu** [comment]
We are also experiencing this in our project, We have put in temporary exclusions for now and will monitor this thread for any changes. https://gitlab.com/crafty-controller/crafty-4/-/commit/ef691eda5c37ae0267cfd99d7d4c48f75d37d984

**Comment by Pierre-Sassoulas** [comment]
It seems like it's a duplicate of https://github.com/pylint-dev/pylint/issues/4987 (pyi support), and should be fixed. Are you using pylint 3 or more, and are you providing the pyi to pylint ? (Disclaimer: I'm not using pyi myself so I don't know for sure if it works, but it *should*).

**Comment by Zedifuu** [comment]
Hi there,

Thank you for the rapid response, It's appreciated.
I'm not sure what you mean by `pyi`, All we've done is a standard install of nh3, use it in our application and then proceed to get the pylint findings, no extra pylint configuration has been applied for nh3.

nh3                           0.2.14
Python                      3.12.1
pip                            23.2.1
pylint                        3.0.3
pylint-gitlab             2.0.0

**Comment by Pierre-Sassoulas** [comment]
> nh3 is Rust but it does have a pyi file which we could potentially start checking.

pylint will need this pyi file to understand the rust code that it can't analyses. (Until recently it couldn't analyses the pyi file either). Or an inference tip that we call brain and that is more internal that that and that we call astroid brain, basically it would mean a pylint plugin for nh3.

**Comment by Zedifuu** [comment]
Oh that's super interesting I also love the naming of asteroid brain 🤩🧠
![image](https://github.com/pylint-dev/pylint/assets/73708675/25086755-36a4-4fe1-b107-541ba0c21618)

OK so I had a root around the nh3 package contents for the pyi file and did indeed find something promising. However I would direct pylint to this file to use it ? Is there configuration in the pylintrc? 

 Apologies if I am referencing something that has already been asked if there is documentation please direct me 😅

**Comment by Pierre-Sassoulas** [comment]
I would make sure that the pyi is accessible by pylint (with the proper namespace like in ``nh3/__init__.py``). I think what you have should work it seems it's in your virtual env or local install..

**Comment by jacobtylerwalls** [comment]
@Pierre-Sassoulas I think for .pyi support we also need #9097, fixed in #9241, waiting on astroid 3.1.

**Comment by Zedifuu** [comment]
Ahh gocha thank you both for the updates <3

**Comment by Pierre-Sassoulas** [comment]
I'm going to try to release astroid 3.1.0 asap then.
